src/renderer/api/endpoints/helm-releases.api.ts

```diff
@@ -155,24 +155,26 @@ export async function rollbackRelease(name: string, namespace: string, revision:
   });
 }
 
-@autobind()
+export interface HelmRelease {
+  appVersion: string;
+  name: string;
+  namespace: string;
+  chart: string;
+  status: string;
+  updated: string;
+  revision: string;
+}
```

Comment:
This seems strange, I know that it is legal but what happens if you import "HelmRelease" somewhere else, does it mean both?

Should we change this to a different name then?

Reply:
Nope, don't we think we should. Class can be merged with interfaces with same name within a module.
I guess it could be considered it as class type extension, but without fields definition in case of `{useDefineForClassFields: true}`

Reply:
What is the benefit of moving this out then? There is `import type { HelmRelease }` already in typescript (see #2777)

Reply:
Consider an example:

```ts
class Test {
  constructor(data: any){
    Object.assign(this, data);
  }
}

class Test2 extends Test {
  public notGood: any;
  
  constructor(data: any){
    super(data);
    console.log(this.notGood); // prints "undefined" when compiled with {useDefineForClassFields: true}
  }
}

console.log(
  new Test2({notGood: true}),
);
```

If we move definition from `Test2` to external interface within class module, then all good and console prints `true`:

```ts
interface Test2 {
 notGood: any;
}
```

Reply:
Wouldn't just not using `Object.assign` also fix this problem?

Reply:
Don't use `Object.assign` at all and remember this choice for every inheriting class? Sounds weird.

Reply:
Exactly, `Object.assign` is weird and is contrary to how Typescript would like you do it. It prevents detecting things like guaranteed assignment in constructors.